### PR TITLE
feat: clean bindings #ntrn-410

### PR DESCRIPTION
**Pull request by NeverHappened**
Remove bindings and related code from neutron-sdk ,neutron-dev-contracts and integration tests
https://hadronlabs.atlassian.net/browse/NTRN-410

PR's:
- [neutron-sdk](https://github.com/neutron-org/neutron-sdk/pull/166) removes bindings and prototypes that are in neutron-std. Use all needed types from neutron-std;
- [neutron-dev-contracts](https://github.com/neutron-org/neutron-dev-contracts/pull/70) replace bindings with grpc
- [neutron-integration-tests](https://github.com/neutron-org/neutron-integration-tests/pull/362) removed binding tests, adapter other things to new dev-contracts changed inteface
- [neutron](https://github.com/neutron-org/neutron/pull/750) some queries and tx's that were used in bindings were missing. Added them.
- [neutron-std](https://github.com/neutron-org/neutron-std/pull/3) regen proto types

### CODE REVIEW

File: x/contractmanager/keeper/msg_server.go [hunk 1]
```diff
@@ -40,3 +40,28 @@ func (k Keeper) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams)
...
+
+// ResubmitFailure resubmits the failure after contract acknowledgement failed
+func (k Keeper) ResubmitFailure(goCtx context.Context, req *types.MsgResubmitFailure) (*types.MsgResubmitFailureResponse, error) {
+	if err := req.Validate(); err != nil {
+		return nil, errors.Wrap(err, "failed to validate MsgUpdateParams")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	sender, _ := sdk.AccAddressFromBech32(req.Sender)
```
Comment (mmikev) [inline x/contractmanager/keeper/msg_server.go]: unhandled error

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
Regenerate swagger, please

File: x/contractmanager/keeper/msg_server.go [hunk 2]
```diff
@@ -40,3 +40,31 @@ func (k Keeper) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams)
...
+func (k Keeper) ResubmitFailure(goCtx context.Context, req *types.MsgResubmitFailure) (*types.MsgResubmitFailureResponse, error) {
+	if err := req.Validate(); err != nil {
+		return nil, errors.Wrap(err, "failed to validate MsgUpdateParams")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	sender, err := sdk.AccAddressFromBech32(req.Sender)
+	if err != nil {
+		return nil, errors.Wrap(sdkerrors.ErrNotFound, "sender in resubmit request is not in correct address format")
```
Comment (mmikev) [inline x/contractmanager/keeper/msg_server.go]: wrong `sdkerrors.ErrNotFound` error type

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

File: x/contractmanager/keeper/msg_server.go [hunk 3]
```diff
@@ -51,8 +51,9 @@ func (k Keeper) ResubmitFailure(goCtx context.Context, req *types.MsgResubmitFai
 
 	sender, err := sdk.AccAddressFromBech32(req.Sender)
 	if err != nil {
-		return nil, errors.Wrap(sdkerrors.ErrNotFound, "sender in resubmit request is not in correct address format")
+		return nil, errors.Wrap(sdkerrors.ErrInvalidAddress, "sender in resubmit request is not in correct address format")
```
Comment (mmikev) [inline x/contractmanager/keeper/msg_server.go]: Why don't you use an error you got from the method?

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

**Review by pr0n00gler — DISMISSED** [review]

File: proto/neutron/contractmanager/tx.proto [hunk 4]
```diff
@@ -43,3 +45,19 @@ message MsgUpdateParams {
...
+message MsgResubmitFailure {
+  option (amino.name) = "contractmanager/MsgResubmitFailure";
+  option (cosmos.msg.v1.signer) = "sender";
+
+  // sender is the contract which failure to acknowledge is resubmitted.
+  string sender = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+
+  // failure_id is id of failure to resubmit
+  //
+  // NOTE: All parameters must be supplied.
```
Comment (swelf19) [inline proto/neutron/contractmanager/tx.proto]: This is a common line for ParamsUpdate methods, and probably copied by mistake, but does not belong here i think

File: x/tokenfactory/keeper/grpc_query.go [hunk 5]
```diff
@@ -44,3 +46,32 @@ func (k Keeper) BeforeSendHookAddress(ctx context.Context, req *types.QueryBefor
...
+	fullDenom, err := types.GetTokenDenom(req.Creator, req.Subdenom)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryFullDenomResponse{FullDenom: fullDenom}, nil
+}
+
+// parseAddress parses address from bech32 string and verifies its format.
+func parseAddress(addr string) (sdk.AccAddress, error) {
```
Comment (swelf19) [inline x/tokenfactory/keeper/grpc_query.go]: Why do you prefer your own custom parser over a general bech32.DecodeAndConvert?

File: x/tokenfactory/keeper/grpc_query.go [hunk 6]
```diff
@@ -44,3 +46,32 @@ func (k Keeper) BeforeSendHookAddress(ctx context.Context, req *types.QueryBefor
...
+	parsed, err := sdk.AccAddressFromBech32(addr)
+	if err != nil {
+		return nil, errors.Wrap(err, "address from bech32")
+	}
+
+	err = sdk.VerifyAddressFormat(parsed)
```
Comment (swelf19) [inline x/tokenfactory/keeper/grpc_query.go]: As far as i can see, verification alreary done in sdk.AccAddressFromBech32 method.

File: proto/osmosis/tokenfactory/v1beta1/query.proto [hunk 7]
```diff
@@ -86,3 +92,14 @@ message QueryBeforeSendHookAddressRequest {
 message QueryBeforeSendHookAddressResponse {
   string contract_addr = 1 [(gogoproto.moretags) = "yaml:\"contract_addr\""];
 }
+
+message QueryFullDenomRequest {
+  string creator = 1 [(gogoproto.moretags) = "yaml:\"creator\""];
+  string subdenom = 2 [(gogoproto.moretags) = "yaml:\"subdenom\""];
+}
+
+// QueryBeforeSendHookAddressResponse defines the response structure for the
```
Comment (swelf19) [inline proto/osmosis/tokenfactory/v1beta1/query.proto]: QueryBeforeSendHookAddressResponse -> QueryFullDenomResponse

File: x/tokenfactory/keeper/grpc_query.go (hunk 5 above)
Comment (NeverHappened) [reply x/tokenfactory/keeper/grpc_query.go]: Don't know. What's the point? I feel DecodeAndConvert kinda more low level.
Also we can just use AccAddressFromBech32 as it includes verification as you pointed out in other comment.

**Review by swelf19 — COMMENTED** [review]
Regenerate swagger pls, to reflect proto description changes

**Comment by NeverHappened** [comment]
@swelf19 done

**Review by swelf19 — DISMISSED** [review]

**Review by pr0n00gler — APPROVED** [review]

File: proto/osmosis/tokenfactory/v1beta1/query.proto [hunk 8]
```diff
@@ -76,13 +82,28 @@ message QueryDenomsFromCreatorResponse {
   repeated string denoms = 1 [(gogoproto.moretags) = "yaml:\"denoms\""];
 }
 
+// QueryBeforeSendHookAddressRequest defines the request structure for the
+// DenomsFromCreator gRPC query.
```
Comment (sotnikov-s) [inline proto/osmosis/tokenfactory/v1beta1/query.proto]:
```suggestion
// QueryBeforeSendHookAddressRequest defines the request structure for the
// BeforeSendHookAddress gRPC query.
```

File: x/contractmanager/keeper/msg_server.go [hunk 9]
```diff
@@ -40,3 +40,32 @@ func (k Keeper) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams)
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+// ResubmitFailure resubmits the failure after contract acknowledgement failed
+func (k Keeper) ResubmitFailure(goCtx context.Context, req *types.MsgResubmitFailure) (*types.MsgResubmitFailureResponse, error) {
```
Comment (sotnikov-s) [inline x/contractmanager/keeper/msg_server.go:45]: why do we have `ResubmitFailure` and `DoResubmitFailure`? what's the difference between them? can we unify so there's only one Keeper's method for failures resubmission?

File: x/contractmanager/keeper/msg_server.go [hunk 10]
```diff
@@ -40,3 +40,32 @@ func (k Keeper) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams)
...
+	if err := req.Validate(); err != nil {
+		return nil, errors.Wrap(err, "failed to validate MsgUpdateParams")
```
Comment (sotnikov-s) [inline x/contractmanager/keeper/msg_server.go]: failed to validate **MsgUpdateParams**?

File: x/contractmanager/keeper/msg_server.go [hunk 11]
```diff
@@ -40,3 +40,32 @@ func (k Keeper) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams)
...
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	sender, err := sdk.AccAddressFromBech32(req.Sender)
+	if err != nil {
+		return nil, errors.Wrap(err, "sender in resubmit request is not in correct address format")
+	}
+
+	if !k.wasmKeeper.HasContractInfo(ctx, sender) {
+		return nil, errors.Wrap(sdkerrors.ErrNotFound, "not a contract address tried to resubmit")
```
Comment (sotnikov-s) [inline x/contractmanager/keeper/msg_server.go]: the error would look like "not a contract address tried to resubmit: not found" with codespace and code referring to sdk.

I think this way it would be more clear. wdyt about something like this?
```go
ErrNotContractResubmission = errors.Register(ModuleName, 1104, "failures resubmission is only allowed to be called by a smart contract")

...
return nil, errors.Wrap(ErrNotContractResubmission, "sender is not a smart contract")
```

would be "sender is not a smart contract: failures resubmission is only allowed to be called by a smart contract" with codespace and code referring to the contractmanager module.

**Review by sotnikov-s — CHANGES_REQUESTED** [review]

File: wasmbinding/message_plugin.go [hunk 12]
```diff
@@ -1065,7 +1065,7 @@ func (m *CustomMessenger) resubmitFailure(ctx sdk.Context, contractAddr sdk.AccA
 		return nil, nil, nil, errors.Wrap(sdkerrors.ErrNotFound, "no failure found to resubmit")
 	}
 
-	err = m.ContractmanagerKeeper.ResubmitFailure(ctx, contractAddr, failure)
+	err = m.ContractmanagerKeeper.DoResubmitFailure(ctx, contractAddr, failure)
```
Comment (mmikev) [inline wasmbinding/message_plugin.go]: We decided to call only MsgServer methods from wasmbindings and no keeper calls, why did you change this?

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

File: wasmbinding/message_plugin.go (hunk 12 above)
Comment (NeverHappened) [reply wasmbinding/message_plugin.go]: Well it was already Keeper call, I just renamed it. I will try to use msg_server method tho

File: x/contractmanager/keeper/msg_server.go (hunk 9 above)
Comment (NeverHappened) [reply x/contractmanager/keeper/msg_server.go:45]: renamed `DoResubmitFailure` to `resubmitFailure` to avoid clashing. This function is keeper function and a little bit lower level dealing with store

File: x/contractmanager/keeper/msg_server.go (hunk 11 above)
Comment (NeverHappened) [reply x/contractmanager/keeper/msg_server.go]: like it!

**Comment by NeverHappened** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/11563490362

File: wasmbinding/message_plugin.go [hunk 13]
```diff
@@ -1084,12 +1084,7 @@ func (m *CustomMessenger) resubmitFailure(ctx sdk.Context, contractAddr sdk.AccA
 		return nil, nil, nil, errors.Wrap(err, "marshal json failed")
 	}
 
-	anyResp, err := types.NewAnyWithValue(failure)
-	if err != nil {
-		return nil, nil, nil, errors.Wrapf(err, "failed to convert {%T} to Any", failure)
-	}
-	msgResponses := [][]*types.Any{{anyResp}}
-	return nil, [][]byte{data}, msgResponses, nil
+	return nil, [][]byte{data}, nil, nil
```
Comment (sotnikov-s) [inline wasmbinding/message_plugin.go]: did you remove msgResponses return on purpose?

**Review by sotnikov-s — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]
